src/openvic-simulation/misc/Modifier.cpp

```diff
@@ -291,23 +303,35 @@ bool ModifierManager::load_triggered_modifiers(ast::NodeCPtr root) {
 key_value_callback_t ModifierManager::_modifier_effect_callback(
 	ModifierValue& modifier, key_value_callback_t default_callback, ModifierEffectValidator auto effect_validator
 ) const {
-
-	return [this, &modifier, default_callback, effect_validator](std::string_view key, ast::NodeCPtr value) -> bool {
-		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
-		if (effect != nullptr) {
-			if (effect_validator(*effect)) {
-				if (!modifier.values.contains(effect)) {
-					return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
-				} else {
-					Logger::error("Duplicate modifier effect: ", key);
-					return false;
-				}
+	std::function<bool(ModifierEffect const*, ast::NodeCPtr)> add_modifier_cb = [this, &modifier, effect_validator](ModifierEffect const* effect, ast::NodeCPtr value) -> bool {
+		if (effect_validator(*effect)) {
+			if (!modifier.values.contains(effect)) {
+				return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
 			} else {
-				Logger::error("Failed to validate modifier effect: ", key);
+				Logger::error("Duplicate modifier effect: ", effect->get_identifier());
 				return false;
 			}
+		} else {
+			Logger::error("Failed to validate modifier effect: ", effect->get_identifier());
+			return false;
 		}
-		return default_callback(key, value);
+	};
+
+	return [this, &modifier, default_callback, effect_validator, add_modifier_cb](std::string_view key, ast::NodeCPtr value) -> bool {
+		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
+		if (effect != nullptr && value->cast_to<ast::IdentifierNode>() != nullptr) {
```

Comment:
```suggestion
		if (effect != nullptr && value->is_type<ast::IdentifierNode>()) {
```
Well done on finding such effective solutions to some very nasty problems, namely that some modifier effect keys have different uses in certain contexts. Here the problem is the list of `colonial_points` provided by a `naval_base` at each level, which causes an error since the list is a different node type from the normal modifier effect value node (Identifier with a decimal value). What's going to be trickier to deal with are the potential cases where the value node type is the same but there's still a special use, as those won't trigger an error but they will load incorrectly.